app/scripts/migrations/054.js

```diff
@@ -0,0 +1,57 @@
+import { cloneDeep } from 'lodash';
+
+const version = 54;
+
+/**
+ * Remove tokens that don't have an address due to
+ * lack of previous addToken validation.  Also removes
+ * an unwanted, undefined image property
+ */
+export default {
+  version,
+  async migrate(originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData);
+    versionedData.meta.version = version;
+
+    const { PreferencesController } = versionedData.data;
+
+    // These properties are arrays and can be directly filtered
+    const tokenArrays = ['tokens', 'hiddenTokens'];
+    tokenArrays.forEach((tokenKey) => {
+      if (PreferencesController[tokenKey]) {
+        PreferencesController[tokenKey] = PreferencesController[
+          tokenKey
+        ].filter(({ address }) => address);
```

Comment:
`hiddenTokens` is an array of strings. If you inspect state, it looks like this:
```
"hiddenTokens": [
  "0x2260fac5e5542a773aa44fbcfedf7c193bc2c599"
],
```
So this filter would filter out all hidden tokens.